### ISSUE TITLE
lein 2.6.1 breaks lein-environ with aot compile.

### ISSUE THREAD

**Issue by EricGebhart** [technomancy/leiningen#2096]
As of this change In leiningen:  https://github.com/technomancy/leiningen/commit/4383caecce94482a57a25146bdd236e89b20f75f

When a project has an :aot compile, instead of writing a  empty `{}`  to the .lein-env file  
lein-environ will write  a file like this.

```#=(clojure.lang.PersistentArrayMap/create {})```

More complicated maps can have even more of that.

A `lein compile` when :aot is present causes environ to blow up when reading with this error.

The explanation for this error is here.  http://dev.clojure.org/jira/browse/CLJ-1138

```
Caused by: java.lang.RuntimeException: No dispatch macro for: =
    at clojure.lang.Util.runtimeException(Util.java:221)
    at clojure.lang.EdnReader$DispatchReader.invoke(EdnReader.java:500)
    at clojure.lang.EdnReader.read(EdnReader.java:142)
    at clojure.lang.EdnReader.read(EdnReader.java:108)
    at clojure.lang.EdnReader.readString(EdnReader.java:64)
    at clojure.edn$read_string.invoke(edn.clj:46)
    at clojure.edn$read_string.invoke(edn.clj:45)
    at environ.core$read_env_file.invoke(core.clj:36)
```

A `lein compile` with the following project will demonstrate the creation of a bad .lein-env file.

`lein new env-bug` 

with this project.clj.

```
(defproject env-bug "0.1.0-SNAPSHOT"
     :description "FIXME: write description"
     :url "http://example.com/FIXME"
     :license {:name "Eclipse Public License"
               :url "http://www.eclipse.org/legal/epl-v10.html"}
     :dependencies [[org.clojure/clojure "1.8.0"]
                    [lein-environ "1.0.2"]]
     :plugins [[lein-environ "1.0.2"]]
     :profiles {:dev {:env {:is-dev true}}}
     :aot [env-bug.core])
```

**Comment by hyPiRion** [comment]
Thanks for the report =)

This should be fixed with #2082, but I'd also think it would be sensible to prevent it in `lein-edn` and `boot-edn` as well. I'll provide a pull request to the repo during the next hours.

**Comment by hyPiRion** [comment]
The next release of environ should have this resolved (weavejester/environ#49), and as such this is effectively a "duplicate" of #2082.

As such I'm closing this if you don't mind. Feel free to reopen if I've overlooked anything.
